src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/TaskService.kt

```diff
@@ -23,7 +35,55 @@ class TaskService(
   private val placementRequestService: PlacementRequestService,
   private val userTransformer: UserTransformer,
   private val placementApplicationService: PlacementApplicationService,
+  private val taskRespository: TaskRespository,
+  private val assessmentRepository: AssessmentRepository,
+  private val placementApplicationRepository: PlacementApplicationRepository,
+  private val placementRequestRepository: PlacementRequestRepository,
 ) {
+  fun getAllReallocatable(
+    allocatedFilter: AllocatedFilter?,
+    page: Int?,
+    sortField: TaskSortField,
+    sortDirection: SortDirection,
+  ): Pair<List<TypedTask>, PaginationMetadata?> {
+    val pageable = getPageable(
+      // Convert to snake_case, because getAllReallocatable is a native SQL query
+      when (sortField) {
+        TaskSortField.createdAt -> "created_at"
+      },
```

Comment:
Optional, but this could be replaced with `sortField.value` if the `TaskSortField` enum is defined in the spec to use snake case for its acceptable values.